gt_concordance/gt_concordance.py

```diff
@@ -0,0 +1,524 @@
+        exit(1)
+    with open(config, 'r') as config_file:
+        config = yaml.load(config_file) or {}
+    context.obj = config
+
+@cli.command()
+@click.pass_context
+def parse_xl_files(context):
+    config = context.obj
+    if is_xl_config_ok(config):
+        files_to_archive = []
+        samples_to_update = []
+
+        # parsing snps file
+        snps_data = parse_maf_snps_file(config)
+
+        # checking xl files
+        XL_FILES_PATH = config.get('XL_FILES_PATH')
+        click.echo('Looking for .xlsx files in {}'.format(XL_FILES_PATH))
+        xl_files = [os.path.join(XL_FILES_PATH, filename) for filename in os.listdir(XL_FILES_PATH)]
+        click.echo('{} files found'.format(len(xl_files)))
+
+        # parsing xl files
+        for xl_file in xl_files:
+            click.echo("Parsing file: {}".format(os.path.basename(xl_file)))
+            xl_file_data = parse_xl_file(config, xl_file)
+            # returns sample name for each created gt file
+            gt_samples = create_gt_files(config, xl_file_data, snps_data, os.path.basename(xl_file))
+            samples_to_update += gt_samples
+            click.echo('{} files have been created in {}/<project>/piper_ngi/03_genotype_concordance'.format(len(gt_samples), config.get('ANALYSIS_PATH')))
+
+            # if ALL gt files were created, archive xl_file
+            if len(gt_samples) == len(xl_file_data):
+                files_to_archive.append(xl_file)
+            # otherwise we keep xl_file in incoming
+            else:
+                click.echo('File will not be archived: {}'.format(xl_file))
+
+        # archive files
+        archived = []
+        XL_FILES_ARCHIVED = config.get('XL_FILES_ARCHIVED')
+        for xl_file in files_to_archive:
+            try:
+                shutil.move(xl_file, XL_FILES_ARCHIVED)
+            except Exception, e:
+                click.echo('Cannot move file {} to {}'.format(xl_file, XL_FILES_ARCHIVED))
+                click.echo(str(e))
+            else:
+                archived.append(xl_file)
+        click.echo('{} files have been archived'.format(len(archived)))
+
+        # update charon
+        for sample in samples_to_update:
+            update_charon(sample, 'AVAILABLE')
+
+def create_gt_files(config, xl_file_data, snps_data, xl_file_name):
+    gt_samples = []
+    for sample_id in xl_file_data:
+        project_id = sample_id.split('_')[0]
+        # create .gt file for each sample
+        output_path = os.path.join(config.get('ANALYSIS_PATH'), project_id, 'piper_ngi/03_genotype_concordance')
+        if not os.path.exists(output_path):
+            click.echo('Output path does not exist! {}'.format(output_path))
+            click.echo('File {}.gt will not be created!'.format(sample_id))
+            continue
+
+        filename = os.path.join(output_path, '{}.gt'.format(sample_id))
+        if os.path.exists(filename):
+            click.echo('File {} already exists. Skipping'.format(filename))
```

Comment:
we discuss about this and the need to stop computaiton in case of real collision between genotypes